ext/mysqli/mysqli.stub.php

```diff
@@ -0,0 +1,548 @@
+
+    /** @return bool */
+    public function multi_query(string $query);
+
+    /** @return bool */
+    public function more_results();
+
+    /** @return bool */
+    public function next_result();
+
+    /** @return bool */
+    public function ping();
+
+    /** @return int|false */
+    public static function poll(?array &$read, ?array &$write, array &$error, int $sec, int $usec = 0);
+
+    /** @return mysqli_stmt|false */
+    public function prepare(string $query);
+
+    /** @return mysqli_result|bool */
+    public function query(string $query, int $resultmode = MYSQLI_STORE_RESULT);
+
+    /** @return mysqli|bool */
+    public function real_connect(
+        ?string $host = null,
+        ?string $user = null,
+        ?string $password = null,
+        ?string $database = null,
+        ?int $port = null,
+        ?string $socket = null,
+        int $flags = 0
+    );
+
+    /** @return string */
+    public function real_escape_string(string $string_to_escape);
+
+    /** @return mysqli_result|bool */
+    public function reap_async_query();
+
+    /** @return string */
+    public function escape_string(string $string_to_escape);
+
+    /** @return bool */
+    public function real_query(string $query);
+
+    /** @return bool */
+    public function release_savepoint(string $name);
+
+    /** @return bool */
+    public function rollback(int $flags = 0, string $name = '');
+
+    /** @return bool */
+    public function savepoint(string $name);
+
+    /** @return bool */
+    public function select_db(string $database);
+
+    /** @return bool */
+    public function set_charset(string $charset);
+
+    /**
+     * @param mixed $value
+     *
+     * @return bool
+     */
+    public function options(int $option, $value);
+
+    /**
+     * @param mixed $value
+     *
+     * @return bool
+     */
+    public function set_opt(int $option, $value);
+
+    /** @return bool */
+    public function ssl_set(
+        string $key,
+        string $cert,
+        string $certificate_authority,
+        string $certificate_authority_path,
+        string $cipher
+    );
+
+    /** @return string|false */
+    public function stat();
+
+    /** @return mysqli_stmt|false */
+    public function stmt_init();
+
+    /** @return mysqli_result|false */
+    public function store_result(int $flags = 0);
+
+    /** @return bool */
+    public function thread_safe();
+
+    /** @return mysqli_result|false */
+    public function use_result();
+
+    /** @return bool */
+    public function refresh(int $options);
+}
+
+class mysqli_warning
+{
+    /**
+     * object $mysqli_link
+     *
+     * @return object|false
+     */
+    public function __construct();
+
+    /** @return bool */
+    public function next();
+}
+
+class mysqli_result
+{
+     /**
+      * mysqli $mysqli_link,
+      * int $resmode = MYSQLI_STORE_RESULT
+      *
+      * @return object|false
+      */
+     public function __construct();
```

Comment:
what kind of errors are thrown when you add the parameters in the definition? 🤔

Reply:
changed and it works with object

Reply:
I get a "Fatal error: Arginfo / zpp mismatch during call of mysqli_stmt::__construct() in Unknown on line 0" when I change the signature to (object $mysqli_link, string $statement)